### ISSUE TITLE
Does it work with http?

### ISSUE THREAD

**Issue by ggerogery** [sciapp/gitlab-registry-cleanup#11]
Hi, 
I've internally hosted a Gitlab instance. And it accessible only through http.

```
>> gitlab-registry-cleanup -g git.a -r registry.git.a -c cred

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 516, in urlopen
    body=body, headers=headers)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 304, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 724, in _validate_conn
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 203, in connect
    conn = self._new_conn()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 133, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 64, in create_connection
    for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
  File "/usr/lib/python3.4/socket.py", line 530, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 362, in send
    timeout=timeout
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 559, in urlopen
    _pool=self, _stacktrace=stacktrace)
  File "/usr/lib/python3/dist-packages/urllib3/util/retry.py", line 245, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/lib/python3/dist-packages/six.py", line 624, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 516, in urlopen
    body=body, headers=headers)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 304, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 724, in _validate_conn
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 203, in connect
    conn = self._new_conn()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 133, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 64, in create_connection
    for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
  File "/usr/lib/python3.4/socket.py", line 530, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
urllib3.exceptions.ProtocolError: ('Connection aborted.', gaierror(-2, 'Name or service not known'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/gitlab-registry-cleanup", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_cleanup/cli.py", line 238, in main
    args.dry_run,
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_cleanup/cli.py", line 223, in cleanup_gitlab_registry
    notify_callback=console_output,
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_cleanup/cleanup.py", line 23, in soft_delete_untagged_imagehashes
    gitlab_registry.delete_image(repository, untagged_imagehash)
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_usage/registry/high_level_api.py", line 188, in delete_image
    self._gitlab_url, self._admin_username, self._admin_auth_token, repository
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_usage/registry/low_level_api.py", line 57, in get_repository_auth_token
    auth_token = _auth_token(repo_auth_url, username, password)
  File "/usr/local/lib/python3.4/dist-packages/gitlab_registry_usage/registry/low_level_api.py", line 31, in _auth_token
    response = requests.get(auth_url, auth=(username, password))
  File "/usr/lib/python3/dist-packages/requests/api.py", line 60, in get
    return request('get', url, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/api.py", line 49, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 457, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 569, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 407, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', gaierror(-2, 'Name or service not known'))
```

**Comment by IngoMeyer441** [comment]
Hey, 
I can add an option to allow plain http. Since the cleanup script runs on the web host itself it should be no security problem. Related to #5.

**Comment by IngoMeyer441** [comment]
I added a `-k` flag to allow plain http in c91ea5f12b6d1f867d0a71a4fd1219c05298e3e3 and released version [v0.1.7 on PyPI](https://pypi.org/project/gitlab-registry-cleanup/0.1.7/). Note that you must write the full qualified url with explicit http protocol:
```bash
gitlab-registry-cleanup -k -g http://git.a -r http://registry.git.a -c cred
```

**Comment by ggerogery** [comment]
Good job, thanks!
